DataViews: centralize the view definition and rename `list` to `table`

oandregal opened the pull request:
Part of https://github.com/WordPress/gutenberg/issues/55083

## What?

This PR:

- centralizes the view definition into a single place
- renames the `list` layout to `table` & updates its icon to `blockTable`
- renames the `side-by-side` layout to `list`

https://github.com/WordPress/gutenberg/assets/583546/e81610c3-077f-435c-9f98-8a10ee6af3ff

## Why?

The centralization helps to visualize the emerging shape of a view object, as well as helping to maintain consistency (see for example how icons drifted due to lacking a single place to update them all: [#56602](https://github.com/WordPress/gutenberg/pull/56602)).

Renaming was part of the tracking issue.

## How?

Centralizes the view definition in `constants.js`.

## Testing Instructions

- Enable the "admin views" experiment and visit "Manage all templates" or/and "Manage all pages".
- Click around everything works as before.

## Follow-ups

- Rename the react view components.

oandregal committed:
ViewTypeSupportsMap: update list to table

oandregal committed:
ViewActions: update list to table

oandregal committed:
Update docs

oandregal committed:
Update CSS selector

oandregal committed:
Replace in the consumers

oandregal committed:
Centralize names in a single place

oandregal committed:
Centralize view definition

oandregal committed:
Centralize icons

oandregal committed:
Pull icon from table

oandregal committed:
Make supports part of VIEW_LAYOUTS

oandregal committed:
Reorganize import from DataViews: import only from directory

oandregal commented on packages/edit-site/src/components/dataviews/constants.js:
```diff
@@ -1,5 +1,53 @@
+export const LAYOUT_GRID = 'grid';
+export const LAYOUT_SIDE_BY_SIDE = 'side-by-side';
+
+export const VIEW_LAYOUTS = [
```
This exercise has shown that there's quite a bit of expectations defined for a view, already. At some point in the following weeks, we could explore having official APIs for some things that were copied around and are now constants.

For example:

- `getIconForViewType( )`
- `viewTypeSupports( 'preview' )`
- etc.

youknowriad reviewed (commented):
I like this PR 👍 Good start. We definitely need things like `getLayout( layout )`. I saw the icon used in the sidebar and it could use such API.

I also think we should start thinking about moving everything to a "dataviews" package. It could contain a store with layout registry for instance...

jameskoster commented:
If 'side-by-side' is to become 'list', then renaming the current 'list' to 'table' seems reasonable. If that's the case we can restore the `blockTable` icon too. Sorry for the detour!

oandregal commented:
> If 'side-by-side' is to become 'list', then renaming the current 'list' to 'table' seems reasonable. If that's the case we can restore the blockTable icon too. Sorry for the detour!

Thanks for the confirmation! I'll do that then.

oandregal committed:
Rename LAYOUT_SIDE_BY_SIDE to LAYOUT_LIST

oandregal committed:
Update LAYOUT_TABLE icon to blockTable

oandregal commented:
ok, this is ready then:

- centralized the view definition into a single place
- renamed the `list` layout to `table` & updates its icon to `blockTable`
- renamed the `side-by-side` layout to `list`

https://github.com/WordPress/gutenberg/assets/583546/e81610c3-077f-435c-9f98-8a10ee6af3ff

One thing I haven't done in this PR is renaming the view components. I'll do that separately so it's easier to review.

ntsekouras reviewed (approved):
LGTM, thank you!

oandregal commented:
follow-up renaming components https://github.com/WordPress/gutenberg/pull/56709
